### PR TITLE
JavaScript: New Lodash Fathead

### PR DESCRIPTION
**Pull request by Lord-Lavios**
## Description of new Instant Answer, or changes
This is my first fathead and my first pull request. IA page can be found here: https://duck.co/ia/view/lodash

The **output.txt** cover some of the redirects and all the articles. How can I test if the output.txt renders correctly because I'm not sure if I've done everything right?

Any feedback is welcome.

## Related Issues and Discussions
#697 

## People to notify
None

## Testing & Review
To be completed by Language Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [x] Title follows correct format (Specifies Instant Answer + Purpose)
- [x] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [x] Instant Answer page is correctly filled out and contains:
    - One topic for the Search Space Language (Java, Python, Scala, Ruby, etc.)
    - One topic from: Reference, Help, Libraries, Tools
        - Documentation Fatheads are considered "Reference"
    - Description, Data source, and 2+ example queries
    - Perl Module (e.g. "DDG::Fathead::PerlDoc" -- we only need a name, not an actual file)
    - Source Name (for "More at <source_name>" link)
    - Source Domain (must contain http:// or https:// -- can be the same as Data Source)
    - Source Info (used as Subtitle for each Article -- usually matches the IA Name)
    - 'Skip Abstract' is checked off
    - Source ID (ping @moollaza to assign one, once Fathead is ready for Beta deploy)

**Code**
- [x] Uniformly indented, well commented
- [x] Fetch.sh and Parse.sh run without errors
- [x] Output contains no blank lines, or multi-line entries
- [ ] Fathead Tests are passing (run `$ duckpan test <fathead_id>`)
    - Tester should report any failures

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/lodash

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/fathead/lodash/parse.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+function getDetails($elem) {
+    var article = $elem.children("h3").attr("id");
+    var url = "https://lodash.com/docs/4.17.4#" + article;
+    var abstractData = {
+        title: $elem.children("h3").children("code"),
+        summary: $elem.children("p").eq(1).text(),
+        args: $elem.children("ol"),
+        returns: $elem.children("p").eq(3),
+        example: $elem.children(".highlight").html()
```
Comment (moollaza) [inline lib/fathead/lodash/parse.js]: @Lord-Lavios you need to grab the plain-text from inside the `.highlight <pre>` -- and put it into `<pre><code>` -- we have JS on the frontend that will apply our own syntax highlighting

Comment (moollaza) [inline lib/fathead/lodash/parse.js]: Until you remove the `<div>` tags, this won't be able to upload to beta, as our processing code tosses out abstracts containing `<div>`

**Comment by Lord-Lavios** [comment]
I've fixed all the bugs that I found. I'll be working on another fathead. Notify me if you find any more bugs.

File: lib/fathead/lodash/trigger_words.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+lodash
+lodash 4.17.4
+lodash javascript
+odash
```
Comment (moollaza) [inline lib/fathead/lodash/trigger_words.txt]: is this a typo?

Comment (moollaza) [inline lib/fathead/lodash/trigger_words.txt]: I think we should add `lodashjs` and maybe `lodash.js`?

File: lib/fathead/lodash/parse.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    var article = $elem.children("h3").attr("id");
+    var url = "https://lodash.com/docs/4.17.4#" + article;
+    var abstractData = {
+        title: $elem.children("h3").children("code"),
+        summary: formatSummary($elem.children("p").eq(1).html()),
+        args: $elem.children("ol"),
+        returns: $elem.children("p").eq(3),
+        example: formatAbstract($elem)
+    };
+    var abstract = ("<section class=\"prog__container\"><pre>" + abstractData.title + "</pre><p>" + abstractData.summary + "</p><span class=\"prog__sub\">Arguments</span><pre>" + abstractData.args + "</pre><span class=\"prog__sub\">Returns</span><pre>" + abstractData.returns + "</pre><span class=\"prog__sub\">Example</span>" + abstractData.example + "</section>").replace(/\t/g, '    ').replace(/\n/g, "\\n");
```
Comment (moollaza) [inline lib/fathead/lodash/parse.js]: @Lord-Lavios please remove the `<pre>` around the `<ol>` -- it's causing the text descriptions of the arguments to be highlighted like code which we don't want.

Also, let's break this down into a few smaller strings that we join -- this line is way too long

Comment (moollaza) [inline lib/fathead/lodash/parse.js]: After that this should be good for merge 👍

File: lib/fathead/lodash/trigger_words.txt (hunk 2 above)
Comment (Lord-Lavios) [inline lib/fathead/lodash/trigger_words.txt]: Fixed it.

File: lib/fathead/lodash/parse.js (hunk 3 above)
Comment (Lord-Lavios) [inline lib/fathead/lodash/parse.js]: I have removed the `<pre>`. I also wanted to break that line into smaller strings but javascript adds extra spaces if I cover the whole in one ES6 string template. So, I've now broken that line into multiple ES6 templates. I've changed this in `script.js` as that is the file used in developing but `parse.js` still shows it as one line.

**Comment by moollaza** [comment]
@Lord-Lavios Thanks! This LGTM :+1:

Excellent work 💯

**Pull request merged**